swift/Workflow/Sources/RenderContext.swift

```diff
@@ -158,3 +152,107 @@ extension RenderContext {
+    }
+
+}
+
+extension Property: AnyWorkflowConvertible {
+
+    public func asAnyWorkflow() -> AnyWorkflow<Value, Never> {
+        return PropertyWorkflow(property: self).asAnyWorkflow()
+    }
+
+
+}
+
+private struct SignalWorkflow<Value>: Workflow {
+    var signal: Signal<Value, Never>
+    typealias Output = Value
+    typealias State = Void
+    func makeInitialState() -> State {
+        return ()
+    }
+    func workflowDidChange(from previousWorkflow: SignalWorkflow, state: inout State) {
+    }
+    typealias Rendering = Void
+    func render(state: State, context: RenderContext<SignalWorkflow>) -> Rendering {
+        let sink = context.makeSink(of: AnyWorkflowAction.self)
+        context.runSideEffect(key: UUID()) { [signal] lifetime in
+            signal
+                .take(during: lifetime)
+                .map { AnyWorkflowAction(sendingOutput: $0) }
+                .observe(on: QueueScheduler.workflowExecution)
+                .observeValues(sink.send)
+        }
+        return ()
+    }
+}
+
+private struct SignalProducerWorkflow<Value>: Workflow {
+    var signalProducer: SignalProducer<Value, Never>
+    typealias Output = Value
+    typealias State = Void
+    func makeInitialState() -> State {
+        return ()
+    }
+    func workflowDidChange(from previousWorkflow: SignalProducerWorkflow, state: inout State) {
+    }
+    typealias Rendering = Void
+    func render(state: State, context: RenderContext<SignalProducerWorkflow>) -> Rendering {
+        let sink = context.makeSink(of: AnyWorkflowAction.self)
+        context.runSideEffect(key: UUID()) { [signalProducer] lifetime in
+            signalProducer
+                .take(during: lifetime)
+                .map { AnyWorkflowAction(sendingOutput: $0) }
+                .observe(on: QueueScheduler.workflowExecution)
+                .startWithValues(sink.send)
+        }
+        return ()
+    }
+}
+
+private struct PropertyWorkflow<Value>: Workflow {
+    var property: Property<Value>
+    typealias Output = Never
+    typealias State = Value
+    func makeInitialState() -> State {
+        return property.value
+    }
+    func workflowDidChange(from previousWorkflow: PropertyWorkflow, state: inout State) {
+    }
+    typealias Rendering = Value
+    func render(state: State, context: RenderContext<PropertyWorkflow>) -> Rendering {
+        let sink = context.makeSink(of: AnyWorkflowAction.self)
+        context.runSideEffect(key: UUID()) { [property] lifetime in
+            property
+                .signal
+                .take(during: lifetime)
+                .map { value in
+                    AnyWorkflowAction { state in
+                        state = value
+                        return nil
+                    }
+                }
+                .observe(on: QueueScheduler.workflowExecution)
+                .observeValues(sink.send)
+        }
+        return property.value
```

Comment:
Should this be `state`?  Otherwise not sure why the state is updated on line 249.

Reply:
Ah, yeah. Should definitely be `state`.